src/audio/rtnr/rtnr.c

```diff
@@ -0,0 +1,554 @@
+#if CONFIG_FORMAT_S16LE
+	{ SOF_IPC_FRAME_S16_LE, rtnr_s16_default },
+#endif // CONFIG_FORMAT_S16LE
+#if CONFIG_FORMAT_S24LE
+	{ SOF_IPC_FRAME_S24_4LE, rtnr_s24_default },
+#endif // CONFIG_FORMAT_S24LE
+#if CONFIG_FORMAT_S32LE
+	{ SOF_IPC_FRAME_S32_LE, rtnr_s32_default },
+#endif// CONFIG_FORMAT_S32LE
+};
+
+const size_t rtnr_fncount = ARRAY_SIZE(rtnr_fnmap);
+
+/*
+ * End of RTNR setup code. Next the standard component methods.
+ */
+
+static struct comp_dev *rtnr_new(const struct comp_driver *drv,
+		struct comp_ipc_config *config,
+		void *spec)
+{
+	struct ipc_config_process *ipc_rtnr = spec;
+	struct comp_dev *dev;
+	struct comp_data *cd;
+	size_t bs = ipc_rtnr->size;
+	int ret;
+
+	comp_cl_info(&comp_rtnr, "rtnr_new()");
+
+	dev = comp_alloc(drv, sizeof(*dev));
+	if (!dev)
+		return NULL;
+
+	dev->ipc_config = *config;
+
+	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*cd));
+	if (!cd)
+		goto err;
+
+	comp_set_drvdata(dev, cd);
+
+	/* Handler for configuration data */
+	cd->model_handler = comp_data_blob_handler_new(dev);
+	if (!cd->model_handler) {
+		comp_cl_err(&comp_rtnr, "rtnr_new(): comp_data_blob_handler_new() failed.");
+		goto err;
+	}
+
+	/* Get configuration data */
+	ret = comp_init_data_blob(cd->model_handler, bs, ipc_rtnr->data);
+	if (ret < 0) {
+		comp_cl_err(&comp_rtnr, "rtnr_new(): comp_init_data_blob() failed.");
+		goto err;
+	}
+
+	/* Component defaults */
+	cd->source_channel = 0;
+	cd->count = 0;
+
+	#if RTNR_ENABLE_PROCESSING
+	RTKMA_API_Context_Create((void *)&cd->rtk_agl);
+	#endif // RTNR_ENABLE_PROCESSING
+
+	/* Done. */
+	dev->state = COMP_STATE_READY;
+	return dev;
+
+err:
+	rfree(dev);
+	rfree(cd);
+	return NULL;
+}
+
+static void rtnr_free(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	comp_info(dev, "rtnr_free()");
+
+	#if RTNR_ENABLE_PROCESSING
+	RTKMA_API_Context_Free((void *)&cd->rtk_agl);
+	free_RtkMemoryPool();
+	#endif // RTNR_ENABLE_PROCESSING
+
+	rfree(cd);
+	rfree(dev);
+}
+
+/* set component audio stream parameters */
+static int rtnr_params(struct comp_dev *dev, struct sof_ipc_stream_params *params)
+{
+	int ret;
+
+	comp_info(dev, "rtnr_params()");
+
+	ret = comp_verify_params(dev, 0, params);
+	if (ret < 0) {
+		comp_err(dev, "rtnr_params() error: comp_verify_params() failed.");
+		return ret;
+	}
+
+	return 0;
+}
+
+static int rtnr_cmd_get_data(struct comp_dev *dev,
+			    struct sof_ipc_ctrl_data *cdata, int max_size)
+{
+	int ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		comp_info(dev, "rtnr_cmd_get_data(), SOF_CTRL_CMD_BINARY");
+		break;
+	default:
+		comp_err(dev, "rtnr_cmd_set_data() error: invalid command %d", cdata->cmd);
+		ret = -EINVAL;
+		break;
+	}
+	return ret;
+}
+
+static int rtnr_cmd_set_data(struct comp_dev *dev,
+			    struct sof_ipc_ctrl_data *cdata)
+{
+	int ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		comp_info(dev, "rtnr_cmd_set_data(), SOF_CTRL_CMD_BINARY");
```

Comment:
you aren't handling the set

Reply:
I found that there is no set/get interface implemented. This is odd and means parameter tuning is not allowed?

Reply:
this looks fixed